sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+using Azure.Core.TestFramework;
+using Azure.Monitor.Query;
+using Azure.Monitor.Query.Models;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NUnit.Framework;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Resources;
+
+#if NET6_0_OR_GREATER
+namespace Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests
+{
+    public class DistroWebAppLiveTests : RecordedTestBase<MonitorExporterTestEnvironment>
+    {
+        private const string TestServerUrl = "http://localhost:9998/";
+
+        // DEVELOPER TIP: Change roleName to something unique when working locally (Example "Test##") to easily find your records.
+        // Can search for all records in the portal via Log Analytics using this query:   Union * | where AppRoleName == 'Test##'
+        private const string RoleName = nameof(DistroWebAppLiveTests);
+
+        private const string LogMessage = "Message via ILogger";
+
+        private LogsQueryClient? _logsQueryClient = null;
+
+        // DEVELOPER TIP: Can pass RecordedTestMode.Live into the base ctor to run this test with a live resource.
+        // DEVELOPER TIP: Can pass RecordedTestMode.Record into the base ctor to re-record the SessionRecords.
+        public DistroWebAppLiveTests(bool isAsync) : base(isAsync) { }
+
+        [RecordedTest]
+        public async Task VerifyDistro()
+        {
+            // SETUP TELEMETRY CLIENT (FOR QUERIYNG LOG ANALYTICS)
+            _logsQueryClient = InstrumentClient(new LogsQueryClient(
+                TestEnvironment.LogsEndpoint,
+                TestEnvironment.Credential,
+                InstrumentClientOptions(new LogsQueryClientOptions()
+                {
+                    Diagnostics = { IsLoggingContentEnabled = true }
+                })
+            ));
+
+            _logsQueryClient.SetQueryWorkSpaceId(TestEnvironment.WorkspaceId);
+
+            // SETUP WEBAPPLICATION WITH OPENTELEMETRY
+            var resourceAttributes = new Dictionary<string, object>
+            {
+                { "service.name", RoleName },
+            };
+
+            var resourceBuilder = ResourceBuilder.CreateDefault();
+            resourceBuilder.AddAttributes(resourceAttributes);
+
+            var builder = WebApplication.CreateBuilder();
+            builder.Logging.ClearProviders();
+            builder.Services.AddOptions<OpenTelemetryLoggerOptions>()
+                .Configure(options =>
+                {
+                    options.SetResourceBuilder(resourceBuilder);
+                });
+            builder.Services.AddOpenTelemetry()
+                .ConfigureResource(x => x.AddAttributes(resourceAttributes))
```

Comment:
Does this invoke `ResourceBuilder.CreateDefault()` or `ResourceBuilder.CreateEmpty()` with attributes? 

Want to understand as the rolename could be different for trace and request/dependency telemetry.

Reply:
I'm calling `CreateDefault()` on line 58. 

In my Asserts below, I'm querying telemetry by the RoleName to confirm that it's the same for all three signals.